packages/flutter/lib/src/material/drawer.dart

```diff
@@ -260,17 +267,22 @@ class Drawer extends StatelessWidget {
       namesRoute: true,
       explicitChildNodes: true,
       label: label,
-      child: ConstrainedBox(
-        constraints: BoxConstraints.expand(width: width ?? drawerTheme.width ?? _kWidth),
-        child: Material(
-          color: backgroundColor ?? drawerTheme.backgroundColor ?? defaults.backgroundColor,
-          elevation: elevation ?? drawerTheme.elevation ?? defaults.elevation!,
-          shadowColor: shadowColor ?? drawerTheme.shadowColor ?? defaults.shadowColor,
-          surfaceTintColor: surfaceTintColor ?? drawerTheme.surfaceTintColor ?? defaults.surfaceTintColor,
-          shape: shape ?? (isDrawerStart
-            ? (drawerTheme.shape ?? defaults.shape)
-            : (drawerTheme.endShape ?? defaults.endShape)),
-          child: child,
+      child: NotificationListener<Notification>(
```

Comment:
This does not feel right, the issue is coming from ScrollNotifications, and this feels a bit too broad.

> When wrapping an the Drawer or NavigationDrawer with own NotificationListener, allowNotifications can enable notification bubbling.

This feels like a foot gun for developers, because they can then reintroduce the issue if they do not stop the notification from bubbling up themselves.

Why not just add a ScrollNotificationObserver here and return true? If folks want to listen to notifications within the drawer, they can subscribe to the observer.

Reply:
That makes sense. I can revert the parameter. Thank you!

It only works when returning `true` with `Notification` for some reason. 
This doesn't work
```dart
      child: ScrollNotificationObserver(
        child: NotificationListener<ScrollNotification>(
          onNotification: (ScrollNotification notification) {
            return true;
          },
          child: ConstrainedBox(
```

This does 
```dart
      child: ScrollNotificationObserver(
        child: NotificationListener<Notification>(
          onNotification: (Notification notification) {
            return true;
          },
          child: ConstrainedBox(
```

Even tho AppBar is looking for scroll notification,  this is a weird.

Reply:
This still stops any and all notifications from leaving the Drawer. I cannot think of anything in particular that would break as a result of this... but I am not sure.
@HansMuller what do you think?

Reply:
I suppose in most cases this would be safe, since the Drawer is usually part of the Scaffold, and usually nothing above the drawer cares if the Drawer is scrolling. I haven't looked at the history of this PR however it seems like it would be safer if the Scaffold squelched scroll notifications from the Drawer, rather than the Drawer doing it. It would still sort-of fail in the oddball case where there are nested scaffolds and was expected that scroll notifications from the inner Scaffold's Drawer would case the outer Scaffold to scroll. That seems like a pretty unlikely use case.

I suppose what we really need here is a way to identify the source of the notification. The AppBar only cares if the body of the Scaffold is scrolling.

Reply:
> I suppose what we really need here is a way to identify the source of the notification. The AppBar only cares if the body of the Scaffold is scrolling.

Notifications could be from any nested children in the body and I don't know if there is a way to filter all notifications from the body. `ScrollNotificationObserver` seems to listen to all scroll notifications. 

Since the app bar should not care about the scrolling notifications from the drawer. 

What If we have a check like this inside the AppBar? This way we don't have squelched scroll notifications from the Drawers and AppBar can don't update `MaterialState.scrolledUnder` if the drawer is open. Wdyt?

```dart
  void _handleScrollNotification(ScrollNotification notification) {
    if (notification is ScrollUpdateNotification && widget.notificationPredicate(notification)) {
      final bool isAnyDrawerOpen = (notification.context?.findAncestorStateOfType<ScaffoldState>()?.isDrawerOpen ?? false)
        || (notification.context?.findAncestorStateOfType<ScaffoldState>()?.isEndDrawerOpen ?? false);
      // If any drawer is open, don't change the scrolledUnder state.
      if (!isAnyDrawerOpen) {
        final bool oldScrolledUnder = _scrolledUnder;
        final ScrollMetrics metrics = notification.metrics;
        switch (metrics.axisDirection) {
          case AxisDirection.up:
            // Scroll view is reversed
            _scrolledUnder = metrics.extentAfter > 0;
            break;
          case AxisDirection.down:
            _scrolledUnder = metrics.extentBefore > 0;
            break;
          case AxisDirection.right:
          case AxisDirection.left:
            // Scrolled under is only supported in the vertical axis, and should
            // not be altered based on horizontal notifications of the same
            // predicate since it could be a 2D scroller.
            break;
        }

        if (_scrolledUnder != oldScrolledUnder) {
          setState(() {
            // React to a change in MaterialState.scrolledUnder
          });
        }
      }
    }
  }
```

Reply:
> Notifications could be from any nested children in the body and I don't know if there is a way to filter all notifications from the body. ScrollNotificationObserver seems to listen to all scroll notifications.

ScrollNotification.context.findAncestorWidgetOfExactType could be used to write a predicate that returned true if the context was a descendant of the Scaffold's body (a predicate like Scaffold.isBodyDescendant(context)).